src/test/java/nustracker/logic/commands/EnrollCommandTest.java

```diff
@@ -1,121 +1,267 @@
-    //                new EnrolledEvents(editedStudent.getEvents().value));
-    //
-    //        String expectedMessage = String.format(EnrollCommand.MESSAGE_ADD_TO_EVENT_SUCCESS, editedStudent);
-    //
-    //        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
-    //        expectedModel.setStudent(firstStudent, editedStudent);
-    //
-    //        assertCommandSuccess(enrollCommand, model, expectedMessage, expectedModel);
-    //    }
-    //
-    //    @Test
-    //    public void execute_invalidStudentIndexUnfilteredList_failure() {
-    //        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredStudentList().size() + 1);
-    //        EnrollCommand enrollCommand = new EnrollCommand(outOfBoundIndex,
-    //                  new EnrolledEvents(VALID_EVENTNAME_TEST));
-    //
-    //        assertCommandFailureShownStudentList(enrollCommand, model,
-    //                                             Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
-    //    }
-    //
-    //    @Test
-    //    public void execute_invalidStudentIndexFilteredList_failure() {
-    //        showStudentAtIndex(model, INDEX_FIRST_STUDENT);
-    //        Index outOfBoundIndex = INDEX_SECOND_STUDENT;
-    //        // ensures that outOfBoundIndex is still in bounds of address book list
-    //        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getStudentList().size());
-    //
-    //        EnrollCommand enrollCommand = new EnrollCommand(outOfBoundIndex,
-    //                  new EnrolledEvents(VALID_EVENTNAME_TEST));
-    //
-    //        assertCommandFailureShownStudentList(enrollCommand, model,
-    //                                             Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
-    //    }
+    @Test
+    public void constructor_nullParameters_throwsNullPointerException() {
+
+        assertThrows(NullPointerException.class, () -> new EnrollCommand(null, null));
+        assertThrows(NullPointerException.class, () -> new EnrollCommand(STUDENTID_ONE, null));
+        assertThrows(NullPointerException.class, () -> new EnrollCommand(null, EVENTNAME_ONE));
+
+    }
+
+    @Test
+    public void execute_enrollStudent_enrollSuccess() {
+        EnrollCommand enrollCommand = new EnrollCommand(STUDENTID_ONE, MATH_OLYMPIAD.getName());
+
+        ModelStubEnrolledEvents compareModel = new ModelStubEnrolledEvents();
+
+        Student currStudent = model.getStudent(STUDENTID_ONE);
+        Event currEvent = model.getEvent(MATH_OLYMPIAD.getName());
+
+        compareModel.addEnrolledEvent(currStudent, currEvent);
+        compareModel.addParticipant(currEvent, STUDENTID_ONE);
+
+        CommandResult actualCommandResult = null;
+
+        try {
+            actualCommandResult = enrollCommand.execute(model, CurrentlyShownList.STUDENTS_LIST);
+        } catch (CommandException e) {
+            fail("Command Exception should not be thrown.");
```

Comment:
I'm not sure what this fail method is, but we have not used it throughout the other test cases. Maybe if not necessary can avoid using it? Unless it works better then go ahead.

Reply:
The fail(String message) method just triggers a fail for the current test with a message when it runs and is one of the assertion methods in JUnit just like assertEquals, so it should be ok to use: https://junit.org/junit5/docs/5.4.0-RC2/api/org/junit/jupiter/api/Assertions.html